scrapers/koji_scraper.py

```diff
@@ -0,0 +1,154 @@
+
+        # Getting tasks for the builds from teiid
+        self.task = self.teiid.query(sql=self.sql_query, retry=3)
+        return self.task
+
+    def get_task_children(self, parent):
+        # SQL query to fetch all child tasks related to a
+        # parent task
+        self.sql_query = """
+                SELECT *
+                FROM brew.task
+                WHERE parent = {}
+                """.format(parent)
+
+        # Getting tasks from teiid
+        self.task = self.teiid.query(sql=self.sql_query, retry=3)
+        return self.task
+
+    def upload_data_to_neo4j(self, builds):
+        # Uploads builds data to their respective nodes
+        log.info('Beginning to upload data to Neo4j')
+        count = 0
+        # Creating the KojiBuild node
+        for build_dict in builds:
+            build = KojiBuild.create_or_update(dict(
+                id_=build_dict['id'],
+                epoch=build_dict['epoch'],
+                state=build_dict['state'],
+                creation_time=build_dict['creation_time'],
+                start_time=build_dict['start_time'],
+                completion_time=build_dict['completion_time'],
+                extra=build_dict['extra'],
+                name=build_dict['package_name'],
+                version=build_dict['version'],
+                release=build_dict['release']
+            ))[0]
+
+            user = User.get_or_create(dict(
+                username=build_dict['owner']
+            ))[0]
+
+            # Defining user Relationships
+            build.owner.connect(user)
+            user.owns_builds.connect(build)
+
+            count += 1
+            log.info('Uploaded {0} builds out of {1}'.format(count, len(builds)))
+
+            if build_dict['task_id']:
+                # Getting task related to the current build
+                task_dict = self.get_corresponding_task(build_dict['task_id'])[0]
+            else:
+                # Continue if the task_id is None
+                continue
+
+            if len(task_dict) == 0:
+                # Continue if no corresponding task found
+                continue
+
+            # Creating KojiTask node
+            task = KojiTask.create_or_update(dict(
+                id_=task_dict['id'],
+                weight=task_dict['weight'],
+                create_time=task_dict['create_time'],
+                start_time=task_dict['start_time'],
+                completion_time=task_dict['completion_time'],
+                state=task_dict['state'],
+                priority=task_dict['priority'],
+                arch=task_dict['arch'],
+                method=task_dict['method']
+            ))[0]
+
+            # Defining Relationships
+            build.tasks.connect(task)
+            user.owns_tasks.connect(task)
+            task.builds.connect(build)
+            task.owner.connect(user)
+
+            if task_dict['parent']:
+                # Getting child tasks of a parent task
```

Comment:
Bike-shedding: This comment isn't very useful since it's obvious that's happening below.